plugins/registry.js

```diff
@@ -308,4 +308,27 @@ export default class PluginRegistry {
 
         return id;
     }
+
+    // Register a component to override file previews. Accepts a function to run before file is
+    // previewed and a react component to be rendered as the file preview.
+    // - handler - A function to check whether preview needs to be overridden. Receives fileInfo and post as arguments.
```

Comment:
0/5 `s/component/render/` - I am not at all sure that it's appropriate in this context

Reply:
Currently, we use `component` for existing methods in webapp reference. I think it fits better. (1/5) E.g.: https://developers.mattermost.com/extend/plugins/webapp/reference/#registerRootComponent